connect/src/com/telenor/connect/AbstractSdkProfile.java

```diff
@@ -96,4 +115,21 @@ public void failure(RetrofitError error) {
                     }
                 });
     }
+
+    @Override
+    public String getLastAuthState() {
+        return lastAuthState;
+    }
+
+    @Override
+    public void validateTokens(ConnectTokensTO tokens, Date serverTimestamp) {
+        Validator.notNullOrEmpty(tokens.getAccessToken(), "access_token");
```

Comment:
I think you are missing fix for one of the issue's requirements:
Do not fail of scope is not present in the /token response.